src/musicbrainz/web/musicbrainzrecordingstask.cpp

```diff
@@ -101,6 +110,9 @@ QNetworkReply* MusicBrainzRecordingsTask::doStartNetworkRequest(
                 << "GET"
                 << networkRequest.url();
     }
+
+    m_measurementTimer.start();
```

Comment:
What this timer doing?!?

Reply:
The timer measures the time form doStartNetworkRequest() until doNetworkReplyFinished(). 
This time is then subtracted from kMinTimeBetweenMbRequests to calculate the required delay value.

Reply:
We cannot make any assumptions on how MusicBrainz has implemented the rate limiting. And also the request might arrive much later their after we have sent them. The safe (and less code) path is to simple wait 1 sec. after receiving the reply as I have implemented it. Otherwise you will see spurious rate limit errors from time to time.

Reply:
And, btw, you would not even need a timer for this purpose. A simple time stamp and calculating the elapsed time afterwards would be sufficient. But as I explained it is neither needed nor desirable.

Reply:
Your concerns are unfounded. Musicbrainz has described that very well and we have confirmed experimental that the first two requests are replied without rate limiting.

Reply:
I don't get it. You add complicated, local code that needs to be maintained instead of using a simple heuristic that is supposed to work most of the time?? Nevertheless, you still need to be prepared to hit the rate limit from time to time due to circumstances out of your control.

Please point me to the statement in the docs that would justify such an implementation!

https://musicbrainz.org/doc/MusicBrainz_API/Rate_Limiting#How_throttling_works
> Currently that rate is (on average) 1 request per second.

This vague statement does not justify any more complex implementation other than a rough 1 second timeout between subsequent requests.

Reply:
"Your concerns are unfounded." is a very objective and thoughtful answer.

Reply:
If you really want to get funky then only apply the rate limiting after hitting it. But an adaptive local timer is useless and not worth the code bloat.

Reply:
I am happy you have found the sentences yourself: 

> Currently that rate is (on average) 1 request per second.

Proposal has an average of 1 s + the average responds time.

Reply:
Safety boundary. Less and simpler code = better.

Implementing a client-side average rate algorithm by making assumptions on how they they measure the average rate internally is pointless. My comments still apply, that doesn't change anything at all.